lib/jxl/modular/encoding/context_predict.h

```diff
@@ -484,13 +485,15 @@ inline PredictionResult Predict(
   size_t offset = 3;
   constexpr bool compute_properties =
       mode & kUseTree || mode & kForceComputeProperties;
-  pixel_type_w left = (x ? pp[-1] : (y ? pp[-onerow] : 0));
-  pixel_type_w top = (y ? pp[-onerow] : left);
-  pixel_type_w topleft = (x && y ? pp[-1 - onerow] : left);
-  pixel_type_w topright = (x + 1 < w && y ? pp[1 - onerow] : top);
-  pixel_type_w leftleft = (x > 1 ? pp[-2] : left);
-  pixel_type_w toptop = (y > 1 ? pp[-onerow - onerow] : top);
-  pixel_type_w toprightright = (x + 2 < w && y ? pp[2 - onerow] : topright);
+  constexpr bool nec = mode & kNoEdgeCases;
```

Comment:
I think it would be a better idea to allocate a slightly larger buffer, and prepare row pointers accordingly... (similarly to how fjxl does it)

Reply:
I'm not sure it is in the long term...

Reply:
Agreed, we should do this at some point. It might be a bit trickier than in fjxl though, because decode-side you cannot do all padding in advance.